replica/table.cc

```diff
@@ -2480,16 +2480,19 @@ write_memtable_to_sstable(reader_permit permit, memtable& mt, sstables::shared_s
 
 future<>
 write_memtable_to_sstable(memtable& mt, sstables::shared_sstable sst, sstables::sstable_writer_config cfg) {
-    return do_with(
-            replica::permit_monitor(make_lw_shared(sstable_write_permit::unconditional())),
-            std::make_unique<reader_concurrency_semaphore>(reader_concurrency_semaphore::no_limits{}, "write_memtable_to_sstable"),
-            cfg,
-            [&mt, sst] (auto& monitor, auto& semaphore, auto& cfg) {
-        return write_memtable_to_sstable(semaphore->make_tracking_only_permit(mt.schema().get(), "mt_to_sst", db::no_timeout, {}), mt, std::move(sst), monitor, cfg)
-        .finally([&semaphore] {
-                return semaphore->stop();
-        });
-    });
+    auto monitor = replica::permit_monitor(make_lw_shared(sstable_write_permit::unconditional()));
+    auto semaphore = std::make_unique<reader_concurrency_semaphore>(reader_concurrency_semaphore::no_limits{}, "write_memtable_to_sstable");
```

Comment:
i think the reason why we were using `semaphore` was that we have to use something moveable . but unfortunately, `reader_concurrency_semaphore` is not. is it possible we can use a plain `reader_concurrency_semaphore` here? as, i think, the C++ runtime can pack it for us in the coroutine frame.

Reply:
@tchaikov is correct, with coroutines, there is no reason anymore to use a unique ptr.